backend/src/test/unimplemented_resources/ID022_collaborationRequestWithMessage.feature

```diff
@@ -0,0 +1,84 @@
+            | id | name       |
+            | 11 | Rust       |
+            | 12 | TypeScript |
+            | 13 | Go         |
+            | 14 | C++        |
+            | 15 | Other      |
+        And the database contains the following urls (ID022):
+            | id | url      |
+            | 16 | test.com |
+        And the database contains the following ideas (ID022):
+            # Date in ms from the unix epoch (Other constructors are deprecated)
+            | id | title             | domains | topics | techs    | date | description | iconUrl | purpose    | author | isPrivate | user |
+            | 17 | Website Idea      | 2       | 5      | 10,11,12 | 1500 | Cool idea   | 15      | Great idea | 1      | False     | 1    |
+            | 18 | Video Game        | 2       | 6      | 13       | 1400 | Cool idea   | 15      | Great idea | 1      | False     | 1    |
+            | 19 | Microcontroller   | 2,4     | 7      | 10,13    | 1300 | Cool idea   | 15      | Great idea | 2      | False     | 2    |
+            | 20 | Command-Line tool | 2       | 8      | 10       | 1200 | Cool idea   | 15      | Great idea | 2      | False     | 2    |
+            | 21 | Novel             | 3       | 9      | 14       | 1100 | Cool idea   | 15      | Great idea | 3      | False     | 3    |
+        And the database contains the following collaboration requests (ID022):
+            | id | ideaId | userId | status   | additionalContact | message                                                                       |
+            | 23 | 17     | 2      | Pending  | "438-764-1940"    | "Hi, I would like to join this project!"                                      |
+            | 24 | 18     | 2      | Declined | null              | "Hi, I have experience in that field and I am interested in working on this!" |
+            | 25 | 20     | 3      | Accepted | null              | "Hi, I have experience in that field and I am interested in working on this!" |
+            | 26 | 21     | 2      | Pending  | null              | "Hi, I have experience in that field and I am interested in working on this!" |
+
+    # Normal/alternate flows
+    Scenario Outline: Successfully send collaboration request
+        Given I am logged in as the user with email "<email>" and password "<password>"
+        When I request to send a collaboration request to the creator of the idea with id "<idea_id>" with message "<message>" and additionalContact "<additionalContact>"
+        Then the number of collaboration requests in the database should be "<total_num_requests>"
+        And the following collaboration request shall exist in the database:
+            | ideaId    | userId    | status  | additionalContact     | message     |
+            | <idea_id> | <user_id> | Pending | "<additionalContact>" | "<message>" |
+        And the number of incoming collaboration requests for the creator of the idea with id "<idea_id>" shall be "<num_incoming_requests>"
```

Comment:
This is really nice but it might be harder to code than just checking that the collaboration request is attached to the right user and the right person. If we have that right we can recover the number of incoming and outgoing requests, but both work, up to you :)